Makefile

```diff
@@ -352,10 +357,12 @@ $(BUILD)/lint: $(LINT_SRC) $(BIN)/revive | $(BUILD)
 # if either changes, this will need to change.
 MAYBE_TOUCH_COPYRIGHT=
 
-$(BUILD)/fmt: $(ALL_SRC) $(BIN)/goimports | $(BUILD)
-	$Q echo "goimports..."
-	$Q # use FRESH_ALL_SRC so it won't miss any generated files produced earlier
-	$Q $(BIN)/goimports -local "github.com/uber/cadence" -w $(FRESH_ALL_SRC)
+# use FRESH_ALL_SRC so it won't miss any generated files produced earlier.
+$(BUILD)/fmt: $(ALL_SRC) $(BIN)/goimports $(BIN)/gofancyimports | $(BUILD)
+	$Q echo "formatting and removing unused imports..."
+	$Q $(BIN)/goimports -w $(FRESH_ALL_SRC)
+	$Q echo "grouping imports..."
+	$Q $(BIN)/gofancyimports fix --local github.com/uber/cadence/ --write $(FRESH_ALL_SRC)
```

Comment:
no strong opinions but I'd also pass `--group-effect`

Reply:
also couldn't see that option but it would be better if ordering of groups were local packages first and then everything else.

Reply:
It's not very configurable, so no - just the built-in groups and not even e.g. "company", just "project and everything else".

It's quite new though, so it seems kinda likely it'll gain more as time goes on.  The Go world has been stuck on goimports for an unbelievably long time, very glad to see a new contender.

Reply:
Re side effects: yeah I kinda lean that way too. They're legitimately special.

Might give it a try tomorrow and see what changes.

Reply:
added that group!